### PR TITLE
ci: move 4.19 complexity tests to tests-datapath-verifier GHA workflow

### PR DESCRIPTION
**Pull request by tklauser**
It seems the verifier issues with 4.19 are resolved when running complexity tests using LVH on that version. Let's cover it in the the GitHub actions workflow instead of Ginkgo tests.

Successful run with test commit: https://github.com/cilium/cilium/actions/runs/4489169184/jobs/7894672779

### CODE REVIEW

File: .github/workflows/tests-datapath-verifier.yaml
Comment (pchaigno) [inline .github/workflows/tests-datapath-verifier.yaml:173]: Did we update the 4.19-main kernel we use here? Is that what we think made the issue disappear?

Comment (tklauser) [reply .github/workflows/tests-datapath-verifier.yaml:173]: I'm not sure how version selection exactly works of the lvh action (/cc @brb), but it seems like we build images from latest 4.19.y stable branch: https://github.com/cilium/little-vm-helper-images/blob/8b568c7421308a200daa53cb595754a7f8e091a0/_data/kernels.json#L23-L26 Current 4.19-main is running 4.19.277:

```
root@complexity-test-4:~# uname -a
Linux complexity-test-4.19 4.19.277 #1 SMP Tue Mar 14 15:39:26 UTC 2023 x86_64 GNU/Linux
```

Comment (brb) [reply .github/workflows/tests-datapath-verifier.yaml:173]: Every merged  little-vm-helper-images PR triggers a rebuild of the LVH VM images which might bump the kernel.

**Review by christarazi — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

**Review by ti-mo — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
:rocket:

**Review by pchaigno — CHANGES_REQUESTED** [review]
Sorry to be a bit of a pain, but this is highly suspicious to me. AFAIK, the complexity issue we were hitting was https://github.com/cilium/cilium/issues/17647#issuecomment-1014678098, which never got fixed on our side and was happening for more than a year.

Did it get fixed in the kernel without us knowing about it? Is there something else that fixed it very recently on our side? Or is this test actually broken again and not testing what we think?

I'll try to check if something recent on the kernel side could explain this.

**Comment by ti-mo** [comment]
> Sorry to be a bit of a pain, but this is highly suspicious to me. AFAIK, the complexity issue we were hitting was [#17647 (comment)](https://github.com/cilium/cilium/issues/17647#issuecomment-1014678098), which never got fixed on our side and was happening for more than a year.
> 
> Did it get fixed in the kernel without us knowing about it? Is there something else that fixed it very recently on our side? Or is this test actually broken again and not testing what we think?
> 
> I'll try to check if something recent on the kernel side could explain this.

No problem at all, we need to know this stuff for sure. Take a look at this: https://github.com/cilium/cilium/actions/runs/4509588303/jobs/7939538892?pr=24538. This has much more detailed output and verifier stats for each tail call. It's based on the changes in this PR (adding 4.19 coverage) and loading all ELFs using Go.

I'll add a uname at the start of the test run so we know exactly what we're dealing with.

**Review by nebril — APPROVED** [review]

**Review by bimmlerd — APPROVED** [review]
CODEOWNERS change looks ok

**Comment by ti-mo** [comment]
@pchaigno Does this really need to remain blocked on your investigation? Sounds like something that can be done on the side. The Go test code looks fine to me. If there are changes to be made to the feature flags, that should be done in a separate PR anyways.

**Comment by ti-mo** [comment]
All tests green and https://github.com/cilium/cilium/actions/runs/4532665447/jobs/7984382192?pr=24538 (test run of https://github.com/cilium/cilium/pull/24538) shows that the largest program (`tail_handle_ipv4_cont`) of the largest bpf_lxc configuration is currently just within the 4.19 limit:

`verifier_test.go:194: tail_handle_ipv4_cont: processed 127506 insns (limit 131072), stack depth 264`

https://github.com/cilium/cilium/issues/17647 mentions a tail call split being merged close after the issue was reported. I see no reason to block this PR any further, so let's merge please.

**Comment by pchaigno** [comment]
I ran a bunch of tests and it looks like it's working as expected. I don't have an explanation for why it was failing only a few weeks/months ago (no significant verifier or datapath changes were merged since then). It could just be small changes pushing us below & above the limit, though that's fairly rare with complexity issues as the complexity doesn't tend to increase/decrease linearly.

That all said, I also checked the workflow is passing and current master and it fails: https://github.com/cilium/cilium/actions/runs/4542358481/jobs/8005682276?pr=24599. That means if we were to merge, we would break CI :disappointed:

**Review by ldelossa — APPROVED** [review]
Approval contingent on: https://github.com/cilium/cilium/pull/24517#issuecomment-1486675725 passing.

**Review by qmonnet — APPROVED** [review]
Putting my docs-structure hat on, and approving again

**Comment by lmb** [comment]
This is :heavy_check_mark:  on 4.19 now, with only cluster aware addressing disabled: https://github.com/cilium/cilium/actions/runs/4609988154/jobs/8147889395

**Review by pchaigno — APPROVED** [review]
LGTM (assuming the test commit is removed of course).

**Review by zacharysarah — APPROVED** [review]
LGTM from a docs perspective. 👍🏻

**Pull request merged**
